server/helpers/validation.ts

```diff
@@ -0,0 +1,70 @@
+import { plainToClass } from 'class-transformer';
+import { validate, ValidationSchema } from 'class-validator';
+import { Request, Response, NextFunction } from 'express';
+import { Post } from '../entities/post';
+
+// As the validation will be put to use on all the classes the structure of this file will need some refactoring.
+
+// conditionalValidation = isOptional -> https://github.com/typestack/class-validator/issues/147
+// The validation still doesn't check what the user inputs very well.
+export const postPUTSchema: ValidationSchema = {
+    name: "postPUTSchema",
+    properties: {
+        category: [{
+            type: 'conditionalValidation', constraints: [(object: any, value: any) => {
+                return object['message'] !== null && object['message'] !== undefined;
```

Comment:
I think `object['message']` should be `object['category']` :D

Reply:
Good eye! I suppose this could be preventable with some unit testing!